NewsFeed/views/CCBRNewsViewController.m

```diff
@@ -83,9 +93,41 @@ - (void)viewDidLoad {
 }
 
 - (void)updateUI {
+    self.loadingIndicatorView.hidden = YES;
     self.collectionView.hidden = self.viewModel.collectionViewHidden;
     self.errorMessageLabel.hidden = self.viewModel.errorMessageLabelHidden;
     [self.collectionView reloadData];
+    [self.collectionView.infiniteScrollingView stopAnimating];
+}
+
+- (void)loadDataError: (NSError*)error {
+    NSString* message = @"API errors";
+    if (![error.localizedDescription isEqualToString:@""]) {
+        message = error.localizedDescription;
+    }
+    if (self.viewModel.collectionViewHidden && !self.viewModel.errorMessageLabelHidden) {
+        self.errorMessageLabel.text = message;
+    } else {
+        [self showMessage:message];
+    }
+    self.loadingIndicatorView.hidden = YES;
```

Comment:
These logic should moved to the View Model. The View-View Controller should be as dump as possible.